caen-watch_acq_times.py

```diff
@@ -0,0 +1,520 @@
+
+def format_config_files(matches: List[str]) -> str:
+    """
+    Format the list of matching config files as a comma-separated string.
+
+    Parameters:
+    matches (List[str]): List of matching config filenames.
+
+    Returns:
+    str: Comma-separated config filenames, or empty string if none.
+    """
+    return ','.join(matches) if matches else ''
+
+def prepare_update_values(
+    sheet: 'GoogleSheet',
+    start_dt: Optional[datetime],
+    stop_dt: Optional[datetime],
+    digitizer: Optional[str],
+    config_files: str
+) -> Dict[int, Any]:
+    """
+    Prepare the dictionary of values to update in the sheet.
+
+    Parameters:
+    sheet (GoogleSheet): The GoogleSheet instance.
+    start_dt (Optional[datetime]): Start time of the run.
+    stop_dt (Optional[datetime]): Stop time of the run.
+    digitizer (Optional[str]): Digitizer info string.
+    config_files (str): Comma-separated config filenames.
+
+    Returns:
+    Dict[int, Any]: Mapping of column indices to values.
+    """
+    return {
+        sheet.COL_SETUP: start_dt,
+        sheet.COL_END: stop_dt,
+        sheet.COL_DAQ_PC: COMPUTER_NAME,
+        sheet.COL_DIGITIZER: digitizer,
+        sheet.COL_CONFIG: config_files
+    }
+
+def warn_if_no_config_matches(matches: List[str], run_name: str) -> None:
+    """
+    Log a warning if no matching config files are found.
+
+    Parameters:
+    matches (List[str]): List of matching config filenames.
+    run_name (str): Name of the run.
+    """
+    if not matches:
+        logging.warning(f"⚠️  No matching config files found for {run_name}")
+
+def process_run_folder(
+    run_name: str,
+    run_folder: Path,
+    sheet: 'GoogleSheet',
+    config_dir: Path,
+    start_dt: Optional[datetime] = None,
+    stop_dt: Optional[datetime] = None,
+    refresh: bool = True
+) -> None:
+    """
+    Process a run folder and atomically update the Google Sheet with all run info.
+
+    Parameters:
+    run_name (str): Name of the run.
+    run_folder (Path): Path to the run folder.
+    sheet (GoogleSheet): GoogleSheet instance for updating the sheet.
+    config_dir (Path): Path to the config directory.
+    start_dt (Optional[datetime]): Start time of the run.
+    stop_dt (Optional[datetime]): Stop time of the run.
+    refresh (bool): Whether to refresh the cache before updating.
+    """
+    settings_path = run_folder / SETTINGS_FILENAME
+    ensure_run_row_exists(sheet, run_name, start_dt, stop_dt, refresh=refresh)
+    digitizer = extract_digitizer_info(str(settings_path))
+    # TEMPORARY OVERRIDE: Map DT5730S (31050) to 'caen8ch' for sheet entry
+    if digitizer == "DT5730S (31050)":
+        digitizer = "caen8ch"
+    matches = find_matching_config_files(str(settings_path), str(config_dir))
+    config_files = format_config_files(matches)
+    values = prepare_update_values(sheet, start_dt, stop_dt, digitizer, config_files)
+    sheet.update_run_row(run_name, values, refresh=refresh)
```

Comment:
We have examples where there are several rows with the same run name. Will this function handle that?